install_config/configuring_extended_ldap_attributes.adoc

```diff
@@ -0,0 +1,175 @@
+section of /etc/sssd/sssd.conf on the authenticating proxy and add this
+attribute:
+
+----
+[domain/example.com]
+...
+ldap_user_extra_attrs = mail
+----
+
+ Next, tell SSSD that it is acceptable for this attribute to be retrieved by
+Apache. Add the following two lines to the *[ifp]* section of
+*_/etc/sssd/sssd.conf_*:
+
+====
+----
+[ifp]
+user_attributes = +mail
+allowed_uids = apache, root
+----
+====
+
+Restart SSSD and test this configuration:
+
+====
+----
+# systemctl restart sssd.service
+
+# getent passwd <username>
+username:*:12345:12345:Example User:/home/username:/usr/bin/bash
+
+# gdbus call \
+        --system \
+        --dest org.freedesktop.sssd.infopipe \
+        --object-path /org/freedesktop/sssd/infopipe/Users/example_2ecom/12345 \
+        --method org.freedesktop.DBus.Properties.Get \
+        "org.freedesktop.sssd.infopipe.Users.User" "extraAttributes"
+(<{'mail': ['username@example.com']}>,)
+----
+====
+
+[[configuring-extended-ldap-attributes-config-apache]]
+== Configuring Apache
+
+Now that SSSD is set up and successfully serving extended attributes, configure
+the web server to ask for them and to insert them in the correct places.
+
+First, install and enable the *mod_lookup_identity* module for Apache:
+
+====
+----
+# yum -y install mod_lookup_identity
+----
+====
+
+Second, enable the module so that Apache will load it. Modify
+*_/etc/httpd/conf.modules.d/55-lookup_identity.conf_* and uncomment the line:
+
+====
+----
+LoadModule lookup_identity_module modules/mod_lookup_identity.so
+----
+====
+
+Next, let SELinux know that it is acceptable for Apache to connect to SSSD over
+D-BUS. Set an SELinux boolean:
+
+====
+----
+# setsebool -P httpd_dbus_sssd on
+----
+====
+
+Then, edit *_/etc/httpd/conf.d/openshift-proxy.conf_* and add the following
+lines inside the *<ProxyMatch /oauth/authorize>* section:
+
+====
+----
+<ProxyMatch /oauth/authorize>
+  AuthName openshift
+
+  LookupOutput Headers <1>
+  LookupUserAttr mail X-Remote-User-Email <1>
+  LookupUserGECOS X-Remote-User-Display-Name <1>
+
+  RequestHeader set X-Remote-User %{REMOTE_USER}s env=REMOTE_USER
+</ProxyMatch>
+----
+<1> Added line.
+====
+
+Restart Apache to pick up the changes:
+
+====
+----
+# systemctl restart httpd.service
+----
+====
+
+[[configuring-extended-ldap-attributes-config-openshift]]
+== Configuring {product-title}
+
+Tell {product-title} where to find these new attributes during login. Edit the
+*_/etc/origin/master/master-config.yaml_* file and add the following lines to
+the *identityProviders* section:
+
+====
+----
+identityProviders:
+ - name: sssd
+ challenge: true
+ login: true
+ mappingMethod: claim
+ provider:
+   apiVersion: v1
+   kind: RequestHeaderIdentityProvider
+   challengeURL: "https://proxy.gallagherhome.com/challenging-proxy/oauth/authorize?${query}"
+   loginURL: "https://proxy.gallagherhome.com/login-proxy/oauth/authorize?${query}"
```

Comment:
Yeah, those were copied straight from my blog (which was incorrectly using my real addresses instead of the example.com ones from the rest of the posts), so I fixed it over there as well.